index.d.ts

```diff
@@ -1,3 +1,9 @@
+type JSONObject = {[key: string]: JSONValue};
```

Comment:
I don't think this does what you intend it to. The way this is defined, no object is considered "JSON stringifyable" unless it has an index signature. For example, this would fail:

```ts
interface Foo { foo: number }
const foo: Foo = { foo: 123 }
const json: JSONObject = foo // Error: Foo is not assignable to JSONObject because it has no index signature
```

I don't think there is currently a way to express this constraint correctly.

Besides, JSON.stringify() actually doesn't care if an object has methods. It will simply omit them (unlike `postMessage()`, which throws an error).

Reply:
This is for the response though, not for being stringifyiable, for stringification we're quite lax: https://github.com/sindresorhus/ky/blob/b6ed9dfc17e1081b992b54f95e65969deb633c5a/index.d.ts#L5